lib/compilers/argument-parsers.js

```diff
@@ -57,6 +57,8 @@ const gccParser = function (compiler) {
             }
             if (options['-fdiagnostics-color']) {
                 if (compiler.compiler.options) compiler.compiler.options += " ";
```

Comment:
As other code might rely on `compiler.options` being populated I think we should move this elsewhere. Perhaps we should default compiler.compiler.options to "" uncondiditionally where they're initialised? I'm not near the source at the moment, so can't suggest a spot.

Reply:
ok i'll take a deeper look